.github/workflows/tics-charm.yaml

```diff
@@ -0,0 +1,41 @@
+name: TICS
+
+on:
+  workflow_call:
+    secrets:
+      TICSAUTHTOKEN:
+        required: true
+
+
+jobs:
+  analyze:
+    runs-on: [self-hosted, linux, amd64, tiobe, jammy]
+
+    steps:
+      - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4
+
+      - name: Install system dependencies
+        run: sudo apt-get update && sudo apt-get install -y python3.10-venv pipx
+
+      - name: Install tox using pipx
+        run: |
+          pipx install tox
+          pipx inject tox tox-uv
+
+      - name: Run tox tests to create coverage.xml
+        run: tox run -e unit
```

Comment:
I'm not necessarily against it but this workflow assumes somethings about how we perform unit tests, making it potentially less reusable. For instance it won't work out-of-the-box with Notary. Shouldn't we have it take the `coverage.xml` as an input instead?

Reply:
I don't love this either. It also requires a small change to the tox even with this (you need to run `coverage report xml`).  

I think you're right, if we can just pass the XML as input, that's much better. My lack of familiarity with github workflows probably has me doing things in a poor way. I'll look into this a bit more.